src/services/gas-price/gas-price.service.ts

```diff
@@ -66,22 +66,24 @@ export class GasPriceService {
   ): Promise<GasSettings> {
     let gasSettings: GasSettings = {};
 
-    const gasLimit = await contractWithSigner.estimateGas[action](
-      ...params,
-      options
-    ).catch(err => {
-      return Promise.reject(err.error);
```

Comment:
here is the problem. there is no error in err

Reply:
Let's avoid changing the structure to try/catch and handle the fact that there might not be an error attribute. Something like:
```ts
return Promise.reject(err?.error || 'Failed to estimate gas');
```

Reply:
what is the purpose of throwing `err?.error`. it seems that in most cases there won't be `error` field in `err` so, we will just throw a general message.

I agree with @agualis and it might be better to remove error handling here until the higher level

Reply:
I think the reason is specifically to catch and throw the error message that comes from the contract failure. How do you know in most cases that the error attribute isn't there?

Reply:
hm, maybe I'm wrong, it just looks weird to me to put `error` field in error 🤔. 

for example in that specific case with RBW / UNIM pool (polygon), in withdraw flow if we add error process `return Promise.reject(err?.error || 'Failed to estimate gas');` we get:

<img width="1500" alt="Screenshot 2023-01-09 at 13 03 30" src="https://user-images.githubusercontent.com/46521087/211272925-365e715d-6f18-4e27-bc07-39fee7259de0.png">

if we remove error process:
<img width="1630" alt="Screenshot 2023-01-09 at 13 01 41" src="https://user-images.githubusercontent.com/46521087/211272130-bd8796a8-0838-4a8e-a74f-74e51114496f.png">

I think second case is more helpful for us as provides more information. 

nevertheless, if we want to handle case with error attribute, I suggest:

`return Promise.reject(err?.error || error);`

Reply:
Ok I see your point, I didn't know exactly how it was working either. Feel free to remove it and let the error be handled higher up the stack.

Reply:
ok. fixed